[ci] remove IS_GHA env var

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #79247
* #79229
* __->__ #79219
* #79214
* #79213
* #79212

This is unnecessary, GitHub automatically populates a `GITHUB_ACTION`
env var:
https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

For docker, this env var is automatically propagated through our use of `--env-file`.

suo committed:
[ci] remove IS_GHA env var

This is unnecessary, GitHub automatically populates a `GITHUB_ACTION`
env var:
https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

For docker, this env var is automatically propagated through our use of `--env-file`.

[ghstack-poisoned]

suo commented:
@pytorchbot merge

malfet commented:
@pytorchbot revert -m "Broke binary jobs see https://hud.pytorch.org/pytorch/pytorch/commit/1a2d95c68a11dd95b4699e7a74d679d8b9a5fb5a" -c nosignal

suo committed:
Update on "[ci] remove IS_GHA env var"


This is unnecessary, GitHub automatically populates a `GITHUB_ACTION`
env var:
https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

For docker, this env var is automatically propagated through our use of `--env-file`.

[ghstack-poisoned]

suo committed:
Update on "[ci] remove IS_GHA env var"


This is unnecessary, GitHub automatically populates a `GITHUB_ACTION`
env var:
https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

For docker, this env var is automatically propagated through our use of `--env-file`.

[ghstack-poisoned]

suo commented:
@pytorchbot merge
